macros/edr/dbt_artifacts/upload_dbt_artifacts.sql

```diff
@@ -1,3 +1,20 @@
-{% macro upload_dbt_artifacts(results) %}
-        {% do elementary.edr_log("Deprecated - Please remove the call to elementary.upload_dbt_artifacts() on your on-run-end hook as it happens automatically now.") %}
+{% macro upload_dbt_artifacts() %}
+  {% if execute and results %}
+    {% set model_upload_func_map = {
+      "dbt_models": elementary.upload_dbt_models,
+      "dbt_tests": elementary.upload_dbt_tests,
+      "dbt_sources": elementary.upload_dbt_sources,
+      "dbt_snapshots": elementary.upload_dbt_snapshots,
+      "dbt_metrics": elementary.upload_dbt_metrics,
+      "dbt_exposures": elementary.upload_dbt_exposures,
+      }
+    %}
+    {% for artifacts_model, upload_artifacts_func in model_upload_func_map.items() %}
+      {% if not elementary.get_result_node(artifacts_model) %}
+        {% do upload_artifacts_func() %}
```

Comment:
I think we should truncate the relation here and not as part of the upload since you call the upload also from the artifact models so no need to do it there (also it's more clear that upload does only the upload)

Reply:
I also think that we should pass the should_commit=True to all the upload functions if you call it from the on-run-end because in redshift it doesn't commit automatically during an on-run-end for every query that you run